rero_ils/modules/holdings/listener.py

```diff
@@ -50,3 +59,75 @@ def enrich_holding_data(sender, json=None, record=None, index=None,
                 break
         json['public_items_count'] = \
             json['items_count'] - number_of_masked_items
+
+
+def update_items_locations_and_types(sender, record=None, **kwargs):
+    """This method checks if the items of the parent record needs an update.
+
+    This method checks the location and item_type of each item attached to the
+    holding record and update the item record accordingly.
+    This method should be connect with 'after_record_update'.
+    :param record: the holding record.
+    """
+    if record.get('$schema') != get_schema_for_resource(Holding):
+        return
+    if record.get('holdings_type') == HoldingTypes.SERIAL:
+        # identify all items records attached to this serials holdings record
+        # with different location and item_type.
+        hold_circ_pid = record.circulation_category_pid
+        hold_loc_pid = record.location_pid
+        search = ItemsSearch().filter('term', holding__pid=record.pid)
+        item_hits = search.\
+            filter('bool', should=[
+                        Q(
+                            'bool', must_not=[
+                                Q('match', item_type__pid=hold_circ_pid)
+                            ]
+                        ),
+                        Q(
+                            'bool', must_not=[
+                                Q('match', location__pid=hold_loc_pid)
+                            ]
+                        )
+            ]
+            ).source(['pid'])
+        items = [hit for hit in item_hits.scan()]
+        item_ids = []
+        # update these items and make sure they have the same location/category
+        # as the parent holdings record.
+        for hit in items:
+            item = Item.get_record_by_pid(hit.pid)
+            if item:
+                item_ids.append(item.id)
+                item_temp_loc_pid, item_temp_type_pid = None, None
+                # remove the item temporary_location if it is equal to the
+                # new item location.
+                temporary_location = item.get('temporary_location', {})
+                if temporary_location:
+                    item_temp_loc_pid = extracted_data_from_ref(
+                        temporary_location.get('$ref'))
+                if hold_loc_pid != item_temp_loc_pid:
+                    if item_temp_loc_pid == hold_loc_pid:
```

Comment:
After `hold_loc_pid != item_temp_loc_pid` ` item_temp_loc_pid == hold_loc_pid` will be never `True`!